swiftide/src/transformers/chunk_code.rs

```diff
@@ -87,14 +87,24 @@ impl ChunkerTransformer for ChunkCode {
     /// - If the code splitting fails, an error is sent downstream.
     #[tracing::instrument(skip_all, name = "transformers.chunk_code")]
     async fn transform_node(&self, node: Node) -> IndexingStream {
+        let file_size = node.chunk.len();
         let split_result = self.chunker.split(&node.chunk);
 
         if let Ok(split) = split_result {
+            let mut offset = 0;
+
             IndexingStream::iter(split.into_iter().map(move |chunk| {
-                Ok(Node {
+                let chunk_size = chunk.len();
+                let mut node = Node {
                     chunk,
                     ..node.clone()
-                })
+                };
+                node.metadata
+                    .insert("Original Size".to_string(), file_size.to_string());
+                node.metadata
+                    .insert("Chunk Offset".to_string(), offset.to_string());
```

Comment:
In all embed modes metadata fields get embedded. So maybe that's not the best idea to put it into metadata. I can imagine though that `original_size: Option<usize>` and `chunk_offset` should just be properties on the node. That also removes the hard dependency with the names in the metadata fields.

Reply:
hmm ok but then we're violating open/closed, should we make a distinction between values that are good for embedding, and values that are not? Perhaps in separate hashmaps?